tests/callbacks/test_stochastic_weight_avg.py

```diff
@@ -61,10 +67,18 @@ def configure_optimizers(self):
             },
         }
 
+    def training_epoch_end(self, _):
+        if not self.crash_after_epoch:
+            return
+        if self.trainer.current_epoch + 1 >= self.crash_after_epoch:
```

Comment:
Note that raising an exception here is a bit silly because the epoch that raised it will be lost since checkpointing has not triggered yet.

The checkpoint loaded will be for the end of the previous epoch.

Reply:
Yes probably makes sense to raise it on the epoch start rather than doing unnecessary work and raising it after the training epoch ends.